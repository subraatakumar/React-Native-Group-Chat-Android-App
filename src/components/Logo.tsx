import React from 'react';
import {View, Image, Text} from 'react-native';
import {ThemeColors} from '../settings/config';
import Logoimg from '../assets/img/logo.gif';
import {deviceHeight, deviceWidth, AppConfig} from '../settings/config';

type LogoPropTypes = {
  dW?: number | undefined;
  dH?: number | undefined;
  title?: string;
};

const Logo = ({
  dH = deviceHeight,
  dW = deviceWidth,
  title = AppConfig.title,
}: LogoPropTypes) => {
  return (
    <View style={{alignItems: 'center', justifyContent: 'center'}}>
      <Image source={Logoimg} style={{width: dW / 3, height: dW / 4.5}} />
      <Text
        style={{
          textAlign: 'center',
          fontSize: dW / 18,
          color: ThemeColors.primary,
          fontWeight: 'bold',
          marginBottom: dW / 20,
        }}>
        {title}
      </Text>
    </View>
  );
};

export default Logo;
