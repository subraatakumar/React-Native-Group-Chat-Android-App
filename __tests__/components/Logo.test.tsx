import React from 'react';
import renderer from 'react-test-renderer';
import Logo from '../../src/components/Logo';

test('Logo renders successfully', () => {
  const logo = renderer.create(<Logo />);
  expect(logo.toJSON()).toMatchSnapshot();
});
