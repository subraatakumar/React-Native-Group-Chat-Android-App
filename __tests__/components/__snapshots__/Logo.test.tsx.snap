// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    source="test-file-stub"
    style={
      Object {
        "height": 166.66666666666666,
        "width": 250,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "rgba(0,128,105,1)",
        "fontSize": 41.666666666666664,
        "fontWeight": "bold",
        "marginBottom": 37.5,
        "textAlign": "center",
      }
    }
  >
    Group Chat
  </Text>
</View>
`;
